pkgs/libresoc/soc.nix

```diff
@@ -20,6 +20,12 @@ with python39Packages;
       fetchSubmodules = false;
     };
 
+    patches = [./soc-nmigen-soc-no-implicit-arg.patch];
```

Comment:
Upstream is using 2 versions of litex, and one of the versions removed this parameter `alignment`, so this patch fixes that

Potentially an upstreamable fix